### PR TITLE
Milestone4: Category Filter

### PR DESCRIPTION
**Pull request by dev1980**
In this milestone the category filter is implemented as follows:
- Enhance the Redux store
- CategoryFilter component was created.
- handleFilterChange method was created.

### CODE REVIEW

File: src/components/CategoryFilter.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+import Proptypes from 'prop-types';
+import categories from '../utils/bookCategories';
+
+const CategoryFilter = ({ changeFilter }) => (
+  <select onChange={e => changeFilter(e.target.value)}>
+    <option>All</option>
+    {categories.map(category => (
+      <option key={category} value={category}>
+        {category}
+      </option>
```
Comment (MauricioRobayo) [inline src/components/CategoryFilter.js]: Nice! You could also combine the 'All' option in the array and then map:

```js
['All', ...categories].map(category => (
```

File: src/reducers/filter.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import actions from '../actions/index';
+
+const { action } = actions;
+const { CHANGE_FILTER } = action;
+
+const filterReducer = (state = '', action) => {
```
Comment (MauricioRobayo) [inline src/reducers/filter.js]: Please make sure the default state for the filter reduce is 'All' [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/ItDefaultsToAll.png)

**Review by MauricioRobayo — CHANGES_REQUESTED** [review]
Nice looking code, it is coming into place, great work!

One minor adjustment before continuing:

* [ ] Please make sure the __`filter.js` reducer__ defaults to 'All' [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone4/ItDefaultsToAll.png).

Status: __Request changes__.

---

__Optional__

There is a warning on the browser console you should take care of:

```
Warning: Use the 'defaultValue' or 'value' props on <select> instead of setting 'selected' on <option>.
```

File: src/components/CategoryFilter.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import Proptypes from 'prop-types';
+import categories from '../utils/bookCategories';
+
+const CategoryFilter = ({ changeFilter }) => (
+  <select onChange={e => changeFilter(e.target.value)}>
+    {['All', ...categories].map(category => (
+      <option key={category} value={category}>
```
Comment (shloch) [inline src/components/CategoryFilter.js:8]: Perfect

File: src/reducers/filter.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    case CHANGE_FILTER:
+      if (action.payload !== 'All') {
+        return action.payload;
+      }
+      return 'All';
+
+    default:
+      return state;
+  }
+};
```
Comment (shloch) [inline src/reducers/filter.js:17]: good job

**Review by shloch — APPROVED** [review]
## good to go
accomplished
### see you on next milestone

**Pull request merged**